src/Config/Validator/Deploy/ReportDirNestingLevel.php

```diff
@@ -0,0 +1,147 @@
+        ResultFactory $resultFactory
+    ) {
+        $this->logger = $logger;
+        $this->environment = $environment;
+        $this->configFileList = $configFileList;
+        $this->file = $file;
+        $this->encoder = $encoder;
+        $this->resultFactory = $resultFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        $envVarValue = $this->environment->getEnvVarMageErrorReportDirNestingLevel();
+        $reportConfigFile = $this->configFileList->getErrorReportConfig();
+
+        try {
+            if (false !== $envVarValue) {
+                $this->logger->notice(sprintf(
+                    'The `%s` environment variable with the value `%s` specifies a custom value for'
+                    . ' the directory nesting level configured for error reporting. This value overrides'
+                    . ' the value specified in the `config.report.dir_nesting_level` property in file %s,'
+                    . ' which will be ignored.',
+                    Environment::ENV_MAGE_ERROR_REPORT_DIR_NESTING_LEVEL,
+                    $envVarValue,
+                    $reportConfigFile
+                ));
+                return $this->resultFactory->success();
+            }
+
+            $configValue = $this->getConfigValue($reportConfigFile);
+            if (null !== $configValue) {
+                $this->logger->notice(sprintf(
+                    'The `config.report.dir_nesting_level` property defined in the file %s with value '
+                    . '`%s` and used to configure the directories nesting level for error reporting.',
+                    $reportConfigFile,
+                    $configValue
+                ));
+                return $this->resultFactory->success();
+            }
+
+            return $this->resultFactory->error(
+                'The directory nesting level value for error reporting has not been configured.',
+                'You can configure the setting using the `config.report.dir_nesting_level`'
+                . ' in the file ' . $reportConfigFile
+            );
+        } catch (FileSystemException $exception) {
+            return $this->resultFactory->error($exception->getMessage());
+        } catch (NotEncodableValueException $exception) {
+            $message = sprintf('Config of the file %s is invalid. ', $reportConfigFile);
+            $message .= $exception->getMessage();
+            return $this->resultFactory->error(
+                $message,
+                'Fix the directory nesting level configuration for error reporting in the file '
+                . $reportConfigFile
+            );
+        }
+    }
+
+    /**
+     * Returns a value of the property `config.report.dir_nesting_level`
+     * from the file  <magento_root>/pub/errors/local.xml
+     *
+     * @param string $file
+     * @return int|string|null
+     * @throws FileSystemException | NotEncodableValueException
```

Comment:
It looks like here is wrong @throws format.
Should be:
```
@throws FileSystemException
@throws NotEncodableValueException
```
https://docs.phpdoc.org/references/phpdoc/tags/throws.html
`It is also RECOMMENDED that this tag occurs for every occurrence of an exception`

Reply:
Done

Reply:
Done